x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -178,51 +132,50 @@ export async function getDocumentSources({
   );
 
   const sourcesWithHasDocs = checkedSources.map((checkedSource) => {
-    const {
-      documentType,
-      hasDocAfter,
-      hasDocBefore,
-      rollupInterval,
-      checkSummaryFieldExists,
-    } = checkedSource;
+    const { documentType, hasDocAfter, hasDocBefore, rollupInterval } =
+      checkedSource;
 
     const hasDocBeforeOrAfter = hasDocBefore || hasDocAfter;
+
     // If there is any data before, we require that data is available before
     // this time range to mark this source as available. If we don't do that,
     // users that upgrade to a version that starts generating service tx metrics
     // will see a mostly empty screen for a while after upgrading.
     // If we only check before, users with a new deployment will use raw transaction
     // events.
     const hasDocs = hasAnySourceDocBefore ? hasDocBefore : hasDocBeforeOrAfter;
+
     return {
       documentType,
       rollupInterval,
-      checkSummaryFieldExists,
       hasDocs,
     };
   });
 
-  const sources: TimeRangeMetadata['sources'] = sourcesWithHasDocs
-    .filter((source) => !source.checkSummaryFieldExists)
-    .map((checkedSource) => {
+  const durationSummarySupport = await getDurationSummarySupportByDocType({
```

Comment:
this is a really quick look so might be wrong, but is this adding a blocking request?

Reply:
Yes. It's a new request. That function needs to be finished before proceeding with the rest of the code.

Reply:
https://www.linkedin.com/posts/tjeerd-schorn-74264a174_meer-dan-450-groei-in-1-jaar-tijd-dit-activity-7140636546671468545-avbo?utm_source=share&utm_medium=member_desktop

Reply:
We need to know the doc types that support the `transaction.duration.summary`, so AFAIK we need a new query to do that. One option could be adding this query to `get_document_availability` `msearch`. Suggestions?

Reply:
I'll move this query up to the `get_document_availability`. This way, there won't be an extra blocking request.